File: api/WebApi/Models/Database/Account/Account.cs
using System;
using System.Collections.Generic;
using System.Linq;
using WebApi.Models.Database.Base;
using WebApi.Models.Domain;

namespace WebApi.Models.Database.Account
{
    public class Account : BaseModel
    {
        public string Name { get; set; }
        public AccountType Type { get; set; }
        public Guid UserId { get; set; }
        public virtual User User { get; set; }

        public double Balance => Transactions?.Sum(t => t.Amount) ?? 0;
        
        public virtual List<Transaction> Transactions { get; set; }
        public virtual List<AccountCategory> Categories { get; set; }
        
        public virtual AccountBudget Budget { get; set; }
        
        public virtual AccountPreferences Preferences { get; set; }

        public Account(string name, AccountType type, Guid userId)
        {
            Name = name;
            Type = type;
            UserId = userId;
        }

    }
}

File: web/src/app/accounts/_models/account.ts
import { AccountPreferences } from './account-preferences';
import { Transaction } from './transaction';
export class Account {
  id: string;
  name: string;
  type: number;
  userId: string;
  balance: number;
  availableBalance: number;
  totalIncome: number;
  totalOutgoings: number;
  transactions: Transaction[];
  preferences: AccountPreferences;
  convertedBalance: number;
  totalConvertedIncome: number;
  totalConvertedOutgoings: number;
}
